### PR TITLE
Add Zeebe secure cluster communication docs

### PR DESCRIPTION
**Pull request by npepinpe**
This PR adds secure cluster communication documentation. It additionally renames the other security pages to make it explicit these are for client/gateway communication, as opposed to inter-cluster communication.

### CODE REVIEW

**Comment by npepinpe** [comment]
I'm afraid it's a little too complicated at the moment, but I kind of ran out of time until my vacations. It can be left as is and I'll pick it up when I'm back, or feel free to rework it or delegate to it to someone else (Chris has previously done the reviews).

I think it would be good to also add a troubleshooting section in the future as we did with the gateway/client communication.

**Comment by npepinpe** [comment]
@Zelldon - let me know if you don't have time for this before the 1.3.0 release (it's fine to review after the code freeze though).

File: docs/self-managed/zeebe-deployment/security/secure-cluster-communication.md
Comment (Zelldon) [inline docs/self-managed/zeebe-deployment/security/secure-cluster-communication.md:81]: Maybe we need to note that it is different to the grpc tls ? We use between client and gateway?

File: docs/self-managed/zeebe-deployment/security/secure-cluster-communication.md
Comment (Zelldon) [inline docs/self-managed/zeebe-deployment/security/secure-cluster-communication.md:118]: Love it thanks for mentioning this

File: docs/self-managed/zeebe-deployment/security/secure-cluster-communication.md
Comment (Zelldon) [inline docs/self-managed/zeebe-deployment/security/secure-cluster-communication.md:124]: Awesome I wanted to ask for an example or link earlier, nice you have added this. :+1:

**Review by Zelldon — DISMISSED** [review]
Love it thanks @npepinpe :bow: Content wise it looks good to me

I think @christinaausley should take a look at the writing again, based on https://confluence.camunda.com/display/HAN/Documentation+Review

**Comment by christinaausley** [comment]
> Love it thanks @npepinpe 🙇 Content wise it looks good to me
> 
> I think @christinaausley should take a look at the writing again, based on https://confluence.camunda.com/display/HAN/Documentation+Review

This is very well-written. I've made minor editorial adjustments in my latest commit and had no large questions about the technical content, but please let me know if you have any questions.

File: docs/self-managed/zeebe-deployment/security/secure-cluster-communication.md
Comment (npepinpe) [reply docs/self-managed/zeebe-deployment/security/secure-cluster-communication.md]: That's already written on the overview page, but it can't hurt to have it here as well, I guess? I'll defer to our editors here @christinaausley - we have two pages, on about client security, and one about cluster security. It's outlined in the security overview page, but I can image users skipping this sometimes and go straight for some link in the sidebar. Do you think we should add a note in both pages, something to the effect of: "If you're looking for client side security, then <link> is the right place", and vice versa.

**Comment by npepinpe** [comment]
@christinaausley - you dismissed the previous review. Is there anything that's blocking this? :slightly_smiling_face:

**Comment by christinaausley** [comment]
@npepinpe LGTM. As I added a commit it must've dismissed the previous reviews.

**Review by deepthidevaki — APPROVED** [review]

**Comment by akeller** [comment]
@npepinpe @menski looks like these went live without redirects for "self-managed/zeebe-deployment/security/authentication" and "self-managed/zeebe-deployment/security/authorization". Can you please add them?

**Comment by menski** [comment]
I pushed the missing redirects in 85de4f704462a9f660f5cc78e4b00e05ec4ba327

**Comment by menski** [comment]
It took me 4 commits, but I hope it's now working as expected

**Pull request merged**
